src/_pytest/nodes.py

```diff
@@ -425,16 +434,19 @@ def add_report_section(self, when: str, key: str, content: str) -> None:
         if content:
             self._report_sections.append((when, key, content))
 
-    def reportinfo(self):
+    def reportinfo(self) -> Tuple[str, Optional[int], str]:
         return self.fspath, None, ""
 
     @property
-    def location(self):
+    def location(self) -> Tuple[str, Optional[int], str]:
         try:
             return self._location
         except AttributeError:
```

Comment:
[`@cached_property`](https://docs.python.org/3/library/functools.html#functools.cached_property) would be nice for this idiom. Maybe we should start using it (with a compat implementation).

Reply:
Thought the same multiple times already.

Reply:
=> https://github.com/pytest-dev/pytest/issues/6131